app/models/miq_expression/field.rb

```diff
@@ -29,11 +30,26 @@ def datetime?
     column_type == :datetime
   end
 
+  def to_s
+    assoc = associations.map { |a| ".#{a}" }.join
+    "#{model.name}#{assoc}-#{column}"
+  end
+
+  def reflections
+    klass = model
+    associations.collect do |association|
+      klass.reflect_on_association(association).tap do |reflection|
+        reflection or raise InvalidFieldError, self
```

Comment:
@kbrock this was `nil` in a previous revision, but leaning more toward raising here. If `#reflections` can return `nil`, `#target` will either fail with a mysterious `nil`, or I'll end up chasing this `nil` down wherever it goes (target ->  arel_attribute, etc..). WDYT?